### PR TITLE
Modified web socket subscription check for given cache entry

### LINKED ISSUE #152: Track subscriptions by key
The problem:

We use [subscription_active](https://github.com/zubr-exchange/cacherpc/blob/master/src/pubsub.rs#L123) method to check whether we can retrieve the value from cache for a particular key. Currently it's implemented by checking AtomicBool flag for a websocket connection to which this key is routed. This flag is updated in [update_status](https://github.com/zubr-exchange/cacherpc/blob/master/src/pubsub.rs#L629) by comparing number of active subscriptions to the number of *desired* subscriptions, which means that if one of subscription confirmations is delayed, all keys routed to this worker are considered not active, and are not retrievable from cache, which is obviously not desirable.

Proposed solution:

Track subscription status by key.

### PR DESCRIPTION
**Pull request by bobs4462**
For a given subscription and commitment level, if a cache entry is
found, and web socket connection is estabilished, then check whether we
have a corresponding active websocket subscription for that
subscription+commitment pair

Closes #152

### CODE REVIEW

**Review by polachok — CHANGES_REQUESTED** [review]
Should be fixed here.

**Comment by bobs4462** [comment]
Added all the requested changes to this branch, the other branch in #185 (which depends on this one) was rebased on top of it.

File: src/rpc.rs [hunk 1]
```diff
@@ -536,13 +548,17 @@ impl Cacheable for GetAccountInfo {
...
+    }
+
     fn is_cacheable(&self, state: &State) -> Result<(), UncacheableReason> {
         if self.config.encoding == Encoding::JsonParsed {
             Err(UncacheableReason::Encoding)
         } else if self.config.data_slice.is_some() {
             Err(UncacheableReason::DataSlice)
-        } else if !state.subscription_active(self.pubkey) {
-            Err(UncacheableReason::Inactive)
+        } else if state.websocket_connected(self.pubkey) {
```
Comment (polachok) [inline src/rpc.rs:560]: maybe `!state.websocket_connected(..)` ?

File: src/rpc.rs [hunk 2]
```diff
@@ -274,8 +274,13 @@ impl State {
         self.accounts.insert(key, data, commitment)
     }
 
-    fn subscription_active(&self, key: Pubkey) -> bool {
-        self.pubsub.subscription_active(key)
+    fn websocket_connected(&self, key: Pubkey) -> bool {
+        self.pubsub.websocket_connected(key)
+    }
+
+    // owner is the subscription for program, if given account belongs to one
```
Comment (polachok) [inline src/rpc.rs:281]: What is this comment about? I don't see any owner around

**Review by polachok — CHANGES_REQUESTED** [review]

File: src/rpc.rs (hunk 1 above)
Comment (bobs4462) [reply src/rpc.rs]: My bad... Fixed

File: src/rpc.rs (hunk 2 above)
Comment (bobs4462) [reply src/rpc.rs]: Sorry, slipped through when I was cherry picking from other branch

**Review by polachok — APPROVED** [review]

**Pull request merged**